parsons/scytl/scytl.py

```diff
@@ -0,0 +1,664 @@
+        self.state = state
+        self.county_name = county_name
+        self.county_election_id = county_election_id
+        self.county_version_num = county_version_num
+        self.county_update_date = county_update_date
+
+
+class Scytl(object):
+    """
+    Instantiate a Scytl connector.
+
+    `Args:`:
+        state: str
+            The two letter code of the state the publishing election results.
+            ex: GA
+        election_id: str
+            The numeric identifier for the election found in the url of the election's website.
+            ex: "114729"
+        county: str (optional)
+            The name of the county publishing the results.
+            ex: Clarke
+    """
+
+    def __init__(self, state: str, election_id: str, county=""):
+        self.state = state
+        self.county = county.replace(' ', '_')
+
+        self.administrator = f"{self.state}/{self.county}" if self.county else self.state
+        self.election_id = election_id
+
+        self.previous_summary_version_num = None
+        self.previous_details_version_num = None
+        self.previous_county_details_version_num = None
+        self.previous_county_details_list = None
+        self.previously_fetched_counties = set([])
+
+    def _parse_date_to_utc(self, input_dt: str):
+        """
+        Parse datetime string as datetime in UTC
+
+        `Args`:
+            input_dt: str
+                The datetime string to be parsed
+        `Returns`:
+            datetime | None
+        """
+
+        if input_dt is None:
+            return None
+
+        temp = parsedate(input_dt, tzinfos=tzinfos)
+        temp = temp.astimezone(timezone('UTC'))
+
+        return temp
+
+    def _get_version(self, administrator: str, election_id: str):
+        """
+        Fetch the latest version of the election results from the Clarity site
+
+        `Args`:
+            administrator: str
+                The url code for the election administrator, either the two-letter
+                state code or the state code and the county, separated by a slash
+            election_id: str
+                The election id for the given election as a string
+        `Returns`:
+            str
+            The version id as a string
+        """
+
+        config_version_url = get_version_url.format(
+            administrator=administrator, election_id=election_id
+        )
+
+        res = requests.get(config_version_url)
+
+        return res.text
+
+    def _parse_file_from_zip_url(self, zipfile_url: str, file_name: str) -> bytes:
+        """
+        Fetch a zip file from the given url and unzip to a byte array
+
+        `Args`:
+            zipfile_url: str
+                The url where the zip file can be found
+            election_id: str
+                The expected name of the file in the zipfile to read
+        `Returns`:
+            bytes
+            The unzipped file as bytes
+        """
+
+        try:
+            zipdata = BytesIO()
```

Comment:
Consider wrapping this in a context manager to automatically handle closing and flushing.

Reply:
I'm not really sure how context managers work, how would they help here?

Reply:
Sorry, I am using a really wonky name. By contextmanager, I mean a `with` expression. 

All IO in python inherits from the IOBase, which supports automatically closing files / freeing memory if you wrap things in `with` statements. 

https://docs.python.org/3/library/io.html#io.IOBase

Here, I think you have a memory leak, since `close()` is never called on `zipdata`.